scripts/Modules/mkw_classes/common.py

```diff
@@ -38,17 +38,55 @@ def __add__(self, other):
     def __sub__(self, other):
         return vec3(self.x - other.x, self.y - other.y, self.z - other.z)
 
+    def __mul__(self, other):
+        """ vec3 * vec3 -> float (dot product)
+            vec3 * float -> vec3 (scalar multiplication)"""
+        if type(other) == vec3:
+            return self.x * other.x + self.y * other.y + self.z * other.z
+        else:
+            return vec3(self.x * other, self.y * other, self.z * other)
+
+    def __matmul__(self, other):
+        """ vec3 @ vec3 -> vec3 (cross product)
+            vec3 @ float -> vec3 (scalar multiplication)"""        
+        if type(other) == vec3:
+            x = self.y*other.z - self.z*other.y
+            y = self.z*other.x - self.x*other.z
+            z = self.x*other.y - self.y*other.x
+            return vec3(x,y,z)
+        else:
+            return vec3(self.x * other, self.y * other, self.z * other)
```

Comment:
Similarly, a designated `cross` method would also likely be better.

Out of curiosity, what is `__matmul__`?

Reply:
The type ambiguity is less of a problem here since matmul always return a vec3 in this context

__matmul__ is a class method, just like __add__ or __mul__. It uses the token "@".
So you write a@b and it's interpreted as a.__matmul__(b)
Here it design the cross product